here-naksha-lib-handlers/src/main/java/com/here/naksha/lib/handlers/internal/IntHandlerForEventHandlers.java

```diff
@@ -63,33 +65,107 @@ public IntHandlerForEventHandlers(final @NotNull INaksha hub) {
     if (pluginValidation instanceof ErrorResult) {
       return pluginValidation;
     }
-    return defaultStorageValidation(eventHandler);
+    return defaultHandlerValidation(eventHandler);
   }
 
-  private Result defaultStorageValidation(EventHandler eventHandler) {
-    if (isDefaultStorageHandler(eventHandler)) {
-      return storageValidationError(eventHandler);
+  private Result defaultHandlerValidation(EventHandler eventHandler) {
+    if (handlerClassMatches(DefaultStorageHandler.class, eventHandler)) {
+      return storageValidationError(eventHandler, DefaultStorageHandlerProperties.STORAGE_ID);
+    }
+    if (handlerClassMatches(DefaultViewHandler.class, eventHandler)) {
+      return propertiesValidationError(eventHandler);
+    }
+    return new SuccessResult();
+  }
+
+  private @NotNull Result propertiesValidationError(EventHandler eventHandler) {
+    Result storageValidation = storageValidationError(eventHandler, DefaultViewHandlerProperties.STORAGE_ID);
+
+    if (!(storageValidation instanceof SuccessResult)) {
+      return storageValidation;
+    }
+
+    Object spaceIdsProperty = eventHandler.getProperties().get(DefaultViewHandlerProperties.SPACE_IDS);
```

Comment:
I think, if we map properties to DefaultViewHandlerProperties upfront, then we will be able to directly code against "spaceIds" as List and will reduce good amount of lines here

Reply:
Updated